presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -38,11 +38,11 @@
 {
     private final ImmutableSet<Rule<?>> rules;
 
-    public PickTableLayout(Metadata metadata)
+    public PickTableLayout(Metadata metadata, SqlParser sqlParser, boolean replaceExistingLayout)
```

Comment:
Can we remove this boolean parameter? I would prefer to have it set to false as it used to be so far, but this would require this rule to be run once. Alternatively, we can evaluate the rule always and rely on `planChanged` method.

Reply:
see comments above.  We want to be able to run this rule more than once before AddExchanges, but the invocation after AddExchanges needs to not overwrite the layout chosen by AddExchanges.

Reply:
realized that we don't need to run after add exchanges, so I removed the boolean parameter

Reply:
instead of removing this flag in a subsequent commit, please do not introduce it all.